dask/array/gufunc.py

```diff
@@ -171,26 +202,108 @@ def apply_gufunc(func, signature, *args, **kwargs):
     if output_sizes is None:
         output_sizes = {}
 
+    ## Axes
+    ### Input validation
```

Comment:
Would it be possible to split some of this logic into another sub-function(s)? `apply_gufunc` is starting to get pretty large.

Reply:
OK - I've refactored the obvious out in a new function called `_validate_normalize_axes` including the relevant tests.